### DISCUSSION TITLE
NER Training on Spacy V3 gets stuck and does not progress further

### DISCUSSION THREAD

**Discussion by shv07** [explosion/spaCy#8157] [Help: Coding & Implementations] [answered]
## How to reproduce the behaviour
I used the standard cli way of training with official config file. The training was for a NER model (non-transformer). 

With GPU the training process gets stuck without progressing further for hours (was at same stage for 4-5 hours). 

Same issue remains after reducing the batch size in the config file - nlp.batch_size = 64 , training.batcher.size = [start - 32, stop - 64]

Screenshot of stage at which the process gets stuck

![image](https://user-images.githubusercontent.com/25014324/118924425-b3d1d300-b95a-11eb-936d-42b078777175.png)

Screenshot of the resources used when the training process gets stuck

![image](https://user-images.githubusercontent.com/25014324/118924149-4aea5b00-b95a-11eb-9f5d-8c3d23d67c38.png)

## Dataset Information:-
* Training Dataset 1 (Works without any errors): data.spacy size -     ,No. of examples -300,000
* Training Dataset 2 (Error): data.spacy size-  ,  800,000 examples
* Validation Set in both cases - 

## Info about spaCy

- **spaCy version:** 3.0.6
- **Platform:** Linux-4.4.0-210-generic-x86_64-with-Ubuntu-16.04-xenial
- **Python version:** 3.6.13
- **Pipelines:** en_core_web_trf (3.0.0), en_core_web_lg (3.0.0)
- **GPU:** 6 GB (Used ~2000MiB /  5942MiB when got stuck)
- **RAM:** 16 GB

**Comment by adrianeboyd** [comment, marked as answer]
It looks like it's thrashing because the train and dev sets are too large to fit into memory. By default, the train set is loaded in memory for easy shuffling, and each time the evaluation step runs (which is what it's probably hanging on here), the full dev set is also loaded into memory.

To train from a large dataset, you'll need to use the `max_epochs = -1` option as described here to stream the train corpus instead, and you'll probably also need a custom corpus reader to handle shuffling and repeating examples: https://spacy.io/usage/training#custom-code-readers-batchers

**Reply by shv07** [reply]
Thanks for the info!
